src/main/java/menion/android/whereyougo/gui/activity/CartridgeDetailsActivity.java

```diff
@@ -86,25 +87,20 @@ public void onCreate(Bundle savedInstanceState) {
 
         tvDistance.setText(Html.fromHtml(buff));
 
-        CustomDialog.setBottom(this, getString(R.string.start), new CustomDialog.OnClickListener() {
-            @Override
-            public boolean onClick(CustomDialog dialog, View v, int btn) {
-                CartridgeDetailsActivity.this.finish();
-                MainActivity.startSelectedCartridge(false);
-                return true;
-            }
-        }, null, null, getString(R.string.navigate), new CustomDialog.OnClickListener() {
-            @Override
-            public boolean onClick(CustomDialog dialog, View v, int btn) {
-                Location loc = new Location(TAG);
-                loc.setLatitude(MainActivity.cartridgeFile.latitude);
-                loc.setLongitude(MainActivity.cartridgeFile.longitude);
-                Guide guide = new Guide(MainActivity.cartridgeFile.name, loc);
-                A.getGuidingContent().guideStart(guide);
-                MainActivity.callGudingScreen(CartridgeDetailsActivity.this);
-                CartridgeDetailsActivity.this.finish();
-                return true;
-            }
+        CustomDialog.setBottom(this, getString(R.string.start), (dialog, v, btn) -> {
```

Comment:
The lambda migration should had to go to #227.
Please do it in a PR with the MainApplication cleanup.